packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,194 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FlushMode } from "@fluidframework/runtime-definitions/internal";
+// The semver package documents and encourages these imports for users that only need some of the semver functionality.
+// eslint-disable-next-line import/no-internal-modules
+import semverGte from "semver/functions/gte.js";
+// eslint-disable-next-line import/no-internal-modules
+import semverLte from "semver/functions/lte.js";
+// eslint-disable-next-line import/no-internal-modules
+import semverValid from "semver/functions/valid.js";
+
+import {
+	disabledCompressionConfig,
+	enabledCompressionConfig,
+} from "./compressionDefinitions.js";
+import { type IContainerRuntimeOptionsInternal } from "./containerRuntime.js";
+import { pkgVersion } from "./packageVersion.js";
+
+/**
+ * Our policy is to support N/N-1 compatibility by default, where N is the most recent public major release of the runtime.
+ * Therefore, if the customer does not provide a compatibility mode, we will default to use N-1.
+ *
+ * However, this is not consistent with today's behavior. Some options (i.e. batching, compression) are enabled by default
+ * despite not being compatible with 1.x clients. Since the policy was introduced during 2.x's lifespan, N/N-1 compatibility
+ * by **default** will be in effect starting with 3.0. Importantly though, N/N-1 compatibility is still guaranteed with the
+ * proper configurations set.
+ *
+ * TODO: This should be updated to "2.0.0" when 3.0 is released.
+ */
+export const defaultCompatibilityMode = "pre-3.0-default";
```

Comment:
can we use something semver compatible here, could we just use today's latest version?

Reply:
The reason I used a non-semver compatible version here was because I didn't want it to overlap with any actual versions. The goal is to maintain today's default configurations if `compatibilityMode` is not defined, but today's default configuratios are inconsistent in terms of what features are enabled/disabled and don't align with _any_ version. For example, there is no such version where batching is enabled, and explicit schema control is disabled. So, the easiest solution IMO is to just use a "placeholder" version until 3.0 is released, then we can use actual semver versions as the default.

Reply:
now that we use a list of versions below can't we just set the exact version of today to the exact values of today, and use that version as the default version?

Reply:
Why can't this just be `2.33.0` and then add entries below to versionDependentOptionConfigMap for `2.33.0` which map for the default values right now. this also gets ride of defaultConfigsForPreFF3

Reply:
Because if a customer would want to actually use `2.33.0` as their compatMode, then they would get the equivalent of `defaultConfigsForPreFF3` which has explicit schema control disabled. If a customer were to use `2.32.0` or `2.34.0`, then explicit schema control would be enabled